From 7645ebdbe282a2c85454037b4d08a5f48f268bc2 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 11 Apr 2017 09:45:42 +0800
Subject: [PATCH 1661/5242] MLK-14655-4 video: mxsfb: check var info during
 initialization for overlay fb

commit  186c0536082e2d6bb351ca844825c498837563bb from
https://source.codeaurora.org/external/imx/linux-imx.git

During overlay fb initialization, check the variable screeninfo
to get some required values for variable screeninfo.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit d2dff36f18a41419055e529bd45d9a264fbc25d5)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 546a489..9f8ae3d 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -2085,6 +2085,7 @@ static void mxsfb_overlay_init(struct mxsfb_info *fbi)
 	}
 
 	/* setup the initial params for overlay fb */
+	overlayfb_check_var(&ofb->ol_fb->var, ofb->ol_fb);
 	overlayfb_set_par(ofb->ol_fb);
 
 	ofb->registered = 1;
-- 
1.7.9.5

